lib/inferno/dsl/input_output_handling.rb

```diff
@@ -169,7 +164,12 @@ def children_available_inputs
       def available_inputs
         @available_inputs ||=
           begin
-            available_inputs = input_definitions
+            available_inputs =
```

Comment:
Having a hard time really understanding this. Could you provide a scenario where `config` will hold inputs that `@inputs` doesn't? My assumption would be that it's because `config` from a group is passed to it's children, even if the children don't need everything in it?

Reply:
Okay I think the above comment helped me figure this out.

Reply:
That is exactly it. Config has to be passed down to children, so some of what it contains will only apply to particular children.